packages/grpc-js/src/server.ts

```diff
@@ -639,6 +639,57 @@ export class Server {
 
       this.sessions.add(session);
 
+      // TODO: Maybe also implement support for grpc.keepalive_permit_without_calls?
+
+      if ('grpc.keepalive_time_ms' in this.options) {
+        // This timeout will be cleaned up automatically as a part of session closing.
```

Comment:
we need to add a max idle timeout so that a connection won't hang forever (if not being closed by client).